packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -674,22 +674,32 @@ export class RunningSummarizer implements IDisposable {
 						});
 						await delay(delaySeconds * 1000);
 					}
-				}
 
-				if (this.shouldAbortOnSummaryFailure) {
-					this.mc.logger.sendTelemetryEvent(
-						{
-							eventName: "ClosingSummarizerOnSummaryStale",
-							reason,
-							message: lastResult?.message,
-						},
-						lastResult?.error,
-					);
+					if (this.shouldAbortOnSummaryFailure) {
+						// There's no need to delay here as we already have summarizeAttemptDelay given to us by the server.
+						if (delaySeconds === 0) {
+							const closeSummarizerDelayOverrideMs =
+								this.mc.config.getNumber(
+									"Fluid.ContainerRuntime.Test.CloseSummarizerOnSummaryStaleDelayOverrideMs",
+								) ?? defaultCloseSummarizerDelayMs;
+							await delay(closeSummarizerDelayOverrideMs);
+						}
+
+						this.mc.logger.sendTelemetryEvent(
+							{
+								eventName: "ClosingSummarizerOnSummaryStale",
+								reason,
+								message: lastResult?.message,
+							},
+							lastResult?.error,
+						);
 
-					this.stopSummarizerCallback("latestSummaryStateStale");
-					this.runtime.closeFn();
-					return;
+						this.stopSummarizerCallback("latestSummaryStateStale");
+						this.runtime.closeFn();
```

Comment:
this.runtime.closeFn(); looks fishy to me. I'm not sure why this block should have slightly different behavior than code couple lines below that also does this.stopSummarizerCallback().

Reply:
In the test, the runtime doesn't close unless I call this here. When I trace the code, there doesn't seem to be any call to close the runtime. I have this code here. It's different, but I am not sure the behavior of the summarizer does what I expect it to do.